scripts/inline_comments_data_collection.py

```diff
@@ -37,57 +34,28 @@ def __init__(self, phid):
         self.phid = phid
 
 
-def find_revision_from_patch(patch_id):
-    diffs = api.search_diffs(diff_id=patch_id)
-
-    if not diffs:
-        raise NoDiffsFoundException(patch_id)
-
-    revision_phid = diffs[0]["revisionPHID"]
-    return revision_phid
-
+def load_revisions_maps():
+    diff_id_to_revision = {}
+    diff_phid_to_id = {}
 
-def find_transactions_from_patch(patch_id):
-    revision_phid = find_revision_from_patch(patch_id)
-    transactions = api.request("transaction.search", objectIdentifier=revision_phid)[
-        "data"
-    ]
-
-    if not transactions:
-        raise NoTransactionsFoundException(patch_id)
-
-    return transactions
+    for revision in phabricator.get_revisions():
+        for transaction in revision["transactions"]:
+            if transaction.get("fields", {}).get("diff") is None:
+                continue
 
+            diff_id_to_revision[transaction["fields"]["diff"]["id"]] = revision
+            diff_phid_to_id[transaction["fields"]["diff"]["phid"]] = transaction[
+                "fields"
+            ]["diff"]["id"]
 
-def get_diff_info_from_phid(phid):
-    diffs = api.search_diffs(diff_phid=phid)
-    if not diffs:
-        raise NoDiffFoundForPHIDException(phid)
-    return diffs[0]["id"], diffs[0]["revisionPHID"]
+    return diff_id_to_revision, diff_phid_to_id
 
 
-def find_details_from_revision_phid(phid):
-    revision = api.load_revision(rev_phid=phid)
+def find_details_from_revision_phid(phid, revisions_map):
+    revision = revisions_map[phid]
     return revision["id"], revision["fields"]["bugzilla.bug-id"]
```

Comment:
This function is no longer used, you could remove it

Reply:
Removed here: 931333c920ee853a98be1e0a2f7bd9d193a31d37